### PR TITLE
[dhctl] GOST digests for dhctl mirror registry copies

### PR DESCRIPTION
**Pull request by mvasl**
## Description

`dhctl mirror` will now compute stribog 256 bit hash for images downloaded to local filesystem. Downloaded image layouts are now stored as tar archive instead of a folder.

Also, dhctl can now detect if it is running outside of container and limit it's functions to only allow usage of `dhctl mirror` subcommand.

## Why do we need it, and what problem does it solve?

We want to be able to digest our registry copies with GOST 34.11-2012

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: feature
summary: dhctl will compute stribog 256 bit hash for downloaded registry copy
impact_level: default
```

### CODE REVIEW

**Comment by mvasl** [comment]
Снимок экрана 2023-11-01 в 23 20 13

File: dhctl/pkg/operations/mirror/bundle.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+func UnpackBundle(mirrorCtx *Context) error {
+	tarFile, err := os.Open(mirrorCtx.TarBundlePath)
+	if err != nil {
+		return fmt.Errorf("read tar bundle: %w", err)
+	}
+
+	tarReader := tar.NewReader(bufio.NewReaderSize(tarFile, 128*1024)) // 128 KiB
+	for {
+		tarHdr, err := tarReader.Next()
```
[the remark this answers is not in the document]
Comment (mvasl) [reply dhctl/pkg/operations/mirror/bundle.go:36]: False positive, sanitation happens within filepath.Join

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/openstack'...

**Comment by mvasl** [comment]
/e2e/abort openstack;Standard;containerd;1.25 6732877404 6732877404-1-con-1-25 /sys/deckhouse-oss/install:pr6409 debian@45.145.65.5

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by mvasl** [comment]
Снимок экрана 2023-11-02 в 15 49 38
Снимок экрана 2023-11-02 в 17 25 07

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Backport failed. See [Job](https://github.com/deckhouse/deckhouse/actions/runs/6808176170) for details.

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [6477](https://github.com/deckhouse/deckhouse/pull/6477) to the branch [release-1.54](https://github.com/deckhouse/deckhouse/tree/release-1.54) successful!

**Pull request merged**